### ISSUE TITLE
Fix all linebreak-style problems not working correctly

### ISSUE THREAD

**Issue by flleeppyy** [eslint/eslint#14309]
**Tell us about your environment**

* **ESLint Version:** 7.23.0
* **Node Version:** 14.16.1
* **npm Version:** 6.14.12
* **Operating System:** Windows 10 Pro w/WSL
* **Visual Studio Code:** 1.55.0

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
@typescript-eslint/parser

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  "env": {
    "browser": true,
    "es2020": true,
    "node": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 12,
    "sourceType": "module"
  },
  "plugins": [
    "@typescript-eslint"
  ],
  
  "rules": {
    "indent": [
      "error",
      2
    ],
    "linebreak-style": [
      "error",
      "unix"
    ],
    "quotes": [
      "error",
      "double"
    ],
    "semi": [
      "error",
      "always"
    ]
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
If I included all the source code for this, the page would be extremely long, and very unneccesary and probably wouldn't help solve the problem.
I got `Expected linebreaks to be 'LF' but found 'CRLF'.` so I did `Ctrl + .` and clicked `Fix all linebreak-style problems`

**What did you expect to happen?**
ESLint to fix all the linebreak style problems

**What actually happened? Please copy-paste the actual, raw output from ESLint.**
ESLint did nothing. Nothing shows in the output in VSCode

**Steps to reproduce this issue:**

1. Have a config where you have a rule for linebreak-style and set it to unix.
2. Open a file with CLRF linebreaks
3. Hover over one of the linebreak-style errors and open Quick Fix, and click Fix all linebreak-style problems
4. Nothing happens

**Are you willing to submit a pull request to fix this bug?**
I have no clue what's happening, so I cannot.

**Comment by mdjermanovic** [comment]
Hi @flleeppyy, thanks for the issue!

> I got `Expected linebreaks to be 'LF' but found 'CRLF'.` so I did `Ctrl + .` and clicked `Fix all linebreak-style problems`
> 
> **What did you expect to happen?**
> ESLint to fix all the linebreak style problems
> 
> **What actually happened? Please copy-paste the actual, raw output from ESLint.**
> ESLint did nothing. Nothing shows in the output in VSCode

Your VSCode is probably set to enforce CRLF, so it sets it back after the ESLint's fix.

Can you try to change this setting in the VSCode status bar (click the CRLF in your status bar, and then select LF)?

![image](https://user-images.githubusercontent.com/44349756/114217186-2321d380-9968-11eb-98aa-58afe884f362.png)

**Comment by flleeppyy** [comment]
I clicked it and set it to LF, but once I switch to another file, it goes back to CLRF

**Comment by flleeppyy** [comment]
Unless I just have to do it for every file? I didn't find a setting that set the default line ending

**Comment by nzakas** [comment]
ESLint isn’t the best way to handle line endings. You can do so in VS Code itself (see https://qvault.io/2020/06/18/how-to-get-consistent-line-breaks-in-vs-code-lf-vs-crlf/) or use a .gitattributes file to have git do it for you.
